src/util/file.cpp

```diff
@@ -1,5 +1,36 @@
 #include "util/file.h"
 
+#include <QRegularExpression>
+
+namespace {
+
+const QRegularExpression kExtractExtensionRegex(R"(\(\*\.(.*)\)$)");
+
+} //anonymous namespace
+
+QString filePathWithSelectedExtension(const QString& fileLocationInput,
+        const QString& fileFilter) {
+    if (fileLocationInput.isEmpty()) {
+        return {};
+    }
+    QString fileLocation = fileLocationInput;
+    if (fileFilter.isEmpty()) {
+        return fileLocation;
+    }
+
+    // Extract 'ext' from QFileDialog file filter string 'Funky type (*.ext)'
+    QRegularExpressionMatch extMatch = kExtractExtensionRegex.match(fileFilter);
+    if (!extMatch.hasMatch()) {
+        return fileLocation;
+    }
+    const QString ext = extMatch.captured(1);
+    QFileInfo fileName(fileLocation);
```

Comment:
```suggestion
    const auto extMatch = kExtractExtensionRegex.match(fileFilter);
    const QString ext = extMatch.captured(1);
    if (ext.isNull()) {
        return fileLocation;
    }
    const QFileInfo fileName(fileLocation)
```

Reply:
> const auto extMatch

The type of auto is not obvious in a review.

Reply:
@Swiftb0y I think `auto` is fine here, but Is checking the string for `isNull()` instead of `isEmpty()` intended?

Reply:
Yes, according to `QRegularExpressionMatch.captured(int)`, if there is no string for the requested capture, the returned string is null. https://doc.qt.io/qt-5/qregularexpressionmatch.html#captured

Not sure if we also want to abort if we got a match, but the match was actually empty.